### ISSUE TITLE
Question:  What settings do I need for AW Spawner blocks

### ISSUE THREAD

**Issue by MohawkyMagoo** [P3pp3rF1y/AncientWarfare2#265]
One of the biggest problems with the AW spawner blocks is that there is no info about them listed anywhere. So any use of them, is entirely trial and error.

What settings do I need to guarantee an INSTANT spawn of the selected mob/s once the player comes within range. Nothing I try seems to work. 

What settings do I need to have it IGNORE nearby entities completely and spawn its mobs regardless?

**Comment by GotoLink** [comment]
To check for player in a given range, set the "required player range" to a strictly positive value.
Set the "debug mode" to false, the spawner will only trigger for non-creative players.

The delay of the first spawn is currently not accessible through GUI.
It is at "max delay" default value, which is 400 server ticks (on "insensible to redstone" setting).
The following spawn tries will occur at a pace randomly distributed between "min delay" and "max delay" range.

To ignore nearby entities, set "max nearby entities" to 0 or less or an extremely large number. (Note in the current code, all entities are accounted for, with the only exception of dropped items...which is something i am willing to change)

**Comment by MohawkyMagoo** [comment]
Oh. All of that explains why my spawners are not working. 

Instant spawning isn't possible at all?  So the min delay is set at 400? by default, and the max delay also at 400? which I'm presuming can be extended upwards? I've been setting spawners at 10 min and 50 max - both of which it seems are forbidden?  Any chance of a quick fix for this - as 400+ spawn times allow the players to break all the spawners before anything can spawn at all.

On entity count - do you mean its counting stuff like torches and other entities, besides mobs/creatures?

**Comment by GotoLink** [comment]
In the current state, it is not impossible to get an instant spawn. It simply takes time.

I think i need to explain that this spawner is different from the vanilla mob spawner in many ways.

The vanilla mob spawner has a default first spawn delay of 20 ticks.
But this delay only start ticking away, if a player (creative included) is within a 16 blocks range.
Its min spawn delay is 200, max spawn delay is 800, max entities is 6 and spawn range is 4 blocks.

On the other hand, the AW spawner starts with 400 ticks, but ticks continuously, no matter if there is a player around or not...as long as the area is loaded for the server, and "redstone/light sensitive" settings are good. Other settings don't matter for ticks. Thus you have a fair amount of time to setup a dungeon, as a creative builder. 

What matters is that once the first delay is done, both calculate the next delay as a random value between min and max spawn delay (200 and 400 for AW, respectively by default).
Meaning that if you let the first 400 ticks go through, by keeping the world loaded, AW spawner will fire next at most with "max spawn delay" value (which is as low as you desire). All those values are even saved within the world, so you don't have to wait all 400 ticks in one go.

Anyway, i will add a setting for the "spawn delay" (which would only be valid for the "next" spawn, after being set, not some weird average stuff)

On entity count, while i didn't checked, it could theoretically include boats, minecarts, fishing hooks, fired snow balls/arrows/splash potions...Torches are either dropped items, or blocks, thus don't count.

**Comment by MohawkyMagoo** [comment]
Hmm, this spawner seems to be very different to the one in 1.6.4 so I'm probably basing assumptions of how it works on something that's completely wrong.

Here how I understood it:

1  If the player is in creative, the spawner does not function at all - this allows creative builds and work, without flooding the area with mobs and wrecking the spawners in the process. So, I'm not sure I understand what the redstone/light sensitive settings are that you talked about above? Working in creative, I should have as long as I need to set up a dungeon, as the spawners should not spawn. or at least that's how they used to be. I never knew what the light sensitive and redstone settings were meant for, and didn't use them, if they existed in 1.6.4.

I designed the Project Zulu spawner with Crudedragos, and that worked really well in 1.6.4 - haven't tried the 1.7.10 version so couldn't say how it performs now. But, the AW spawner was created by myself and Shadowmage as a mini-copy of it, since Shadowmage didn't want any other mod dependency, and the AW scanner would not register it anyway.

The basics that are really required of a mob spawner for Aw structures are fairly simple -

Capability of instantly spawning the contents when a player comes within range.
Capability of spawning the contents as a rolling release, to the max limit set, as a vanilla spawner would.
Single spawns that replace the spawner block - not random position spawns.

Spawners should only activate - when a player comes within activation range - I'm really not sure what is achieved by having them start on chunk load? Does that just mean all a spawners contents would spawn, whether a player activates it or not? Or are they just ready to activate?

**Comment by GotoLink** [comment]
This is simply an optimization feature.
All vanilla mob spawners simply check each tick the entire player list to see if any is within activation range. Then they wait the delay, then they do other checks to spawn group of mobs. Which means that the more players, the more spawners, the more server cpu time is spent on trying to activate mob spawners, multiplicatively.
The AW "Advanced spawners" wait the delay, then check players within range (if any is set), then do the other checks to spawn a group of mobs.
Since a delay is much faster to process than player and other checks, it is much more optimized.
And with a randomized delay, the load of the other checks are diluted over the blocks, making a much smoother game.
Most checks are also optional, to further improve the efficiency of each spawner.

Anyways, i am working on it right this instant.

**Comment by MohawkyMagoo** [comment]
Ahh, ok, I understand. Thanks for the explanation. Was experimenting with the spawner set ups as it stands at the moment. Setting nearby monsters to 0 results in spawner not spawning anything at all. Will try again with a huge number and see if that works.

EDIT: So I tried again, this time with the spawners set to 100 for nearby monsters. Still nothing spawning.

This is a sample spawner set up. Can you spot anything that's obviously wrong?

rule:
plugin=awTownHall
number=102
data:
JSON:{{"val":{"teData":{"val":{"id":{"val":"advanced_spawner_tile","id":"ss"},"spawnerSettings":{"val":{"spawnDelay":{"val":"400","id":"pi"},"debugMode":{"val":"0","id":"pb"},"respondToRedstone":{"val":"0","id":"pb"},"minDelay":{"val":"200","id":"pi"},"inventory":{"val":{"itemList":{"val":[],"id":"ls"}},"id":"ct"},"lightSensitive":{"val":"0","id":"pb"},"playerRange":{"val":"6","id":"pi"},"maxNearbyMonsters":{"val":"100","id":"pi"},"maxDelay":{"val":"400","id":"pi"},"transparent":{"val":"1","id":"pb"},"spawnGroups":{"val":[{"val":{"settingsList":{"val":[{"val":{"remainingSpawnCount":{"val":"8","id":"pi"},"entityId":{"val":"ProjectZulu|Core.PZBat","id":"ss"},"minToSpawn":{"val":"4","id":"pi"},"maxToSpawn":{"val":"8","id":"pi"}},"id":"ct"}],"id":"ls"},"groupWeight":{"val":"1","id":"pi"}},"id":"ct"}],"id":"ls"},"xpToDrop":{"val":"0","id":"pi"}},"id":"ct"},"z":{"val":"1616","id":"pi"},"y":{"val":"110","id":"pi"},"x":{"val":"1803","id":"pi"}},"id":"ct"},"blockName":{"val":"AncientWarfareStructure:advanced_spawner","id":"ss"},"meta":{"val":"0","id":"pi"},"buildPass":{"val":"0","id":"pi"}},"id":"ct"}}
:enddata
:endrule

Setting "Invulnerability" (so block cannot be broken) to -1 does not work, value isn't accepted.

Also, thought you might like to see the old screen caps from the PZ mob spawner which show its functionality. These were used when Shadowmage made his copy.

![capture1](https://cloud.githubusercontent.com/assets/2426129/7104392/b01b4bc4-e0d5-11e4-8cd1-4e13fb5df7b5.JPG)
This is the first screen on opening the GUI. The user has much the same choices as the AW spawner, but its simpler to use. The one thing it did have was the spawn offsets from the spawner, so that individual spawns could be location controlled. 
Clicking on "New Entry" takes you to this screen:
![capture2](https://cloud.githubusercontent.com/assets/2426129/7104398/29765810-e0d6-11e4-8955-966bff81d055.JPG)
The "Name" box fills with the name of the mob you select by clicking on the button on the right of it, which brings up this screen:
![capture3](https://cloud.githubusercontent.com/assets/2426129/7104399/5b428d00-e0d6-11e4-9a6a-6da8c731a59f.JPG)
The menu bar that has appeared on the right is the folder system for every mod the user has loaded that adds mobs or creatures - this means you can select what you want quite quickly by clicking on the folder, in this case Grimoire of Gaia has been selected:
![capture4](https://cloud.githubusercontent.com/assets/2426129/7104400/9c26199a-e0d6-11e4-9dd9-9a305cda766d.JPG)
The user then selects what mob he wants by clicking on it. 
![capture5](https://cloud.githubusercontent.com/assets/2426129/7104402/b895e966-e0d6-11e4-8045-903571d275e0.JPG)
The rest of the control settings for the mob - number, delays etc are pretty much the same as AW's current spawner.
The thing it has that was incredibly useful was the ability to play a sfx when the spawner activated and spawned its mobs. See the "Soud" box? Clicking on the right hand button brought up a folder menu of all sfx from mobs that the user had loaded, plus and custom folders the user may have created (I used quite a few)
![capture6](https://cloud.githubusercontent.com/assets/2426129/7104408/258381fa-e0d7-11e4-9539-66576d081fd6.JPG)
Clicking on the sfx parent directory allowed the user to select a sound from a mod or a custom one. 
![capture7](https://cloud.githubusercontent.com/assets/2426129/7104410/4dd4f1ac-e0d7-11e4-8f47-bd97fc4ed245.JPG)
Further settings allowed for things like "play once on spawn" "loop" etc etc.

On the whole it was a very robust system and worked really well. This is what we were trying to recreate for AW.

The differences in use between the old PZ one and the AW one are huge in terms of time taken. Before Shadowmage left, I ran comparative tests on stocking the same structure using the two different spawners, on average it took about ten minutes to get an effective dungeon set up using PZ. With AW, because of the way mobs are jumbled in the selection screen, it was taking over an hour and frequently more. That's a lot of time when stocking a lot of structures, and most of it spent pointlessly scrolling through the selection list looking for the mob you want.

The AW spawner in some respects is more advanced than the old PZ one, and has some great extra features that PZ did not. 

But it lost the simplicity and especially the speed of use that the PZ one possessed.

**Comment by MohawkyMagoo** [comment]
On the whole selection list scrolling issue - my current set up gives me 800+ entries of mobs and creatures and entities for selection. This can be reduced by about a hundred using the skippables list in config to delete entries which should not be used in the mob spawner. (I will post you a new one on github - the one in the default for the mod is way out of date) 

But heres the list as it looks, that I have to scroll through to find the one mob I'm looking for. 

This gets more complicated as the scroll bar itself gets narrower and narrower the more mobs there are to select - so actually being able to press on it at all is a game all by itself. (Same problem on structure selection GUI when selecting a structure to spawn in creative)

Heres the mobs list - you can see the problem:
(Consider also that there are four big mob mods missing from this list too - doesn't include Lycanites, Dungeon Mobs, Twilight Forest, Oceancraft, Atum etc. You would probably double this list size.)

primitivemobs.RocketCreeper
primitivemobs.Minion
AncientWarfareNpc.viking.soldier
npchumanfemale
AncientWarfareNpc.native.archer.elite
MoCreatures.Wyvern
MoCreatures.Mole
GrimoireOfGaia.Flesh Lich
lom_TRex
AncientWarfareNpc.native.mounted_archer
Enderman
ProjectZulu|Core.PZBat
AncientWarfareNpc.aw_npc_priest
AncientWarfareNpc.pirate.archer
primitivemobs.HauntedTool
AncientWarfareNpc.custom_2.archer.elite
Megalania
Komodo
lotr.UrukWarg
lotr.RangerNorthCaptain
lotr.AngmarOrcTrader
chocolateQuest.SummonedBait
lotr.DolGuldurBannerBearer
npcfurryfemale
chocolateQuest.CQ_npc
fossil.Spinosaurus
AncientWarfareNpc.bandit.leader.elite
GrimoireOfGaia.Valkyrie
npcskeleton
lotr.WoodElf
lotr.DunlendingBannerBearer
BiomesOPlenty.Bird
GrimoireOfGaia.Anubis
witchery.illusionZombie
lotr.RohanBannerBearer
lotr.HobbitBartender
AncientWarfareNpc.custom_1.mounted_archer
AncientWarfareNpc.bandit.trader
witchery.corpse
GrimoireOfGaia.Harpy
lotr.NearHaradDrinksTrader
lom_HammerShark
witchery.illusionCreeper
ProjectZulu|Core.Eagle
AncientWarfareNpc.bandit.archer
fossil.Elasmotherium
AncientWarfareNpc.custom_2.bard
npcvillager
lotr.HobbitOrcharder
GrimoireOfGaia.Cyclops
primitivemobs.FestiveCreeper
Pig
npcfurrymale
lotr.MordorWarg
MoCreatures.PetScorpion
GSZombieSkullCrawler
AncientWarfareNpc.bandit.priest
AncientWarfareNpc.custom_3.civilian.male
AncientWarfareNpc.desert.soldier.elite
witchery.villageguard
AncientWarfareNpc.aw_npc_combat
witchery.owl
witchery.familiar
MoCreatures.KomodoDragon
lotr.HighElf
MoCreatures.Deer
ProjectZulu|Core.RipperFin
chocolateQuest.abyssWalker
GrimoireOfGaia.Ender Dragon Girl
lotr.MirkTroll
lom_Squirrel
lotr.GaladhrimElf
lotr.MordorBannerBearer
fossil.Pachycephalosaurus
lotr.GondorRuinsWraith
AncientWarfareNpc.custom_1.civilian.female
AncientWarfareNpc.desert.leader.elite
lotr.BlueDwarfMiner
lotr.GaladhrimTrader
AncientWarfareNpc.custom_1.soldier
GSZombieCat
Silverfish
AncientWarfareNpc.custom_3.soldier.elite
GrimoireOfGaia.Trader
lom_Bison
AncientWarfareNpc.custom_2.archer
ExtendedFamPack.Familiar Cube
AncientWarfareNpc.bandit.soldier
fossil.Quagga
lotr.HobbitShirriffChief
AncientWarfareNpc.custom_2.leader.elite
AncientWarfareNpc.viking.cavalry
Squid
Skeleton
MoCreatures.Kitty
lotr.Dunlending
lotr.UtumnoIceWarg
MoCreatures.SmallFish
chocolateQuest.pirateBoss
MoCreatures.Elephant
primitivemobs.SheepmanSmith
LeashKnot
MoCreatures.JellyFish
Wolf
lotr.DunlendingArcher
lotr.RangerNorth
lotr.GaladhrimWarrior
lotr.Dwarf
chocolateQuest.monking
lom_HermitCrab
MoCreatures.Boar
lotr.MordorOrcArcher
BiomesOPlenty.Glob
npcorcfemale
LavaMonsters.LavaMonster
primitivemobs.MimicChest
lotr.Crocodile
lotr.AngmarOrcWarrior
witchery.werevillager
witchery.mindrake
GrimoireOfGaia.Dullahan
fossil.Allosaurus
lotr.AngmarWarg
GrimoireOfGaia.Wither Cow
Tortoise
fossil.TRex
ProjectZulu|Core.Green Finch
witchery.illusionSpider
MoCreatures.MediumFish
lotr.UtumnoOrc
AncientWarfareNpc.bandit.civilian.female
AncientWarfareNpc.viking.archer.elite
AncientWarfareNpc.custom_3.cavalry
fossil.Dodo
ProjectZulu|Core.Elephant
chocolateQuest.greenDragon
AncientWarfareNpc.custom_3.bard
CustomNpc
AncientWarfareNpc.native.bard
ProjectZulu|Core.Zebra
AncientWarfareNpc.desert.leader
PigZombie
MoCreatures.Crab
lotr.NearHaradrimWarlord
AncientWarfareNpc.viking.civilian.female
AncientWarfareNpc.desert.soldier
ProjectZulu|Core.Haunted Armor
ProjectZulu|Core.GiantRat
lom_Ammonite
AncientWarfareNpc.custom_3.leader.elite
AncientWarfareNpc.custom_2.soldier
ProjectZulu|Core.Polar Bear
fossil.Liopleurodon
MoCreatures.Rat
MoCreatures.Ray
NpcDragon
lotr.RohirrimMarshal
lom_Narwal
lotr.DunlendingBartender
lotr.RohanBlacksmith
ProjectZulu|Core.ThrowingRock
MoCreatures.Maggot
GrimoireOfGaia.Siren
MoCreatures.Bear
lotr.NearHaradMerchant
primitivemobs.BabySpider
MoCreatures.Mouse
AncientWarfareNpc.viking.leader
AncientWarfareNpc.native.trader
MoCreatures.Piranha
Bat
VillagerGolem
npcorcfmale
lotr.UrukHaiSapper
MoCreatures.Ogre
GrimoireOfGaia.Yuki-Onna
chocolateQuest.pirate
AncientWarfareNpc.custom_2.soldier.elite
piranha
lotr.DwarfBannerBearer
Zombie
lotr.HobbitDrunkard
lotr.Boar
lotr.HighElfBannerBearer
LittleTurtle
AncientWarfareNpc.viking.trader
lom_Raptor
MoCreatures.Firefly
fossil.Deinonychus
chocolateQuest.minotaur
AncientWarfareNpc.custom_3.leader
GrimoireOfGaia.Sahuagin
AncientWarfareNpc.aw_npc_worker
lotr.MordorSpider
chocolateQuest.spiderBoss
lotr.MordorWargBombardier
primitivemobs.Sheepman
AncientWarfareNpc.bandit.soldier.elite
ProjectZulu|Core.Lizard
lotr.UrukHaiBerserker
chocolateQuest.specter
primitivemobs.BlazingJuggernaut
ProjectZulu|Core.Brown Bear
fossil.DodoEgg
lotr.DunlendingWarrior
LavaSlime
ProjectZulu|Core.Centipede
ProjectZulu|Core.Mammoth
lotr.Banner
lotr.Rohirrim
primitivemobs.BrainSlime
AncientWarfareNpc.native.priest
lotr.Lion
lotr.AngmarOrcMercenaryCaptain
witchery.goblinmog
AncientWarfareNpc.viking.mounted_archer
AncientWarfareNpc.desert.cavalry
witchery.item
GrimoireOfGaia.Cobblestone Golem
lotr.ShirePony
lom_Shark
chocolateQuest.mechaHeavy
GrimoireOfGaia.Sphinx
AncientWarfareNpc.custom_1.cavalry
GrimoireOfGaia.Vampire
lotr.GaladhrimBannerBearer
Chicken
ProjectZulu|Core.Minotaur
chocolateQuest.Hookshoot
npcdwarfmale
lotr.Gemsbok
GrimoireOfGaia.Centaur
Slime
npcdwarffemale
AncientWarfareNpc.viking.leader.elite
witchery.spectre
lom_Triceratops
witchery.imp
AncientWarfareNpc.bandit.cavalry
GrimoireOfGaia.Jorogumo
chocolateQuest.dummy
lotr.DesertScorpion
MoCreatures.Duck
AncientWarfareNpc.custom_1.trader
ProjectZulu|Core.BeetleBS
lotr.UtumnoOrcArcher
GrimoireOfGaia.Selkie
WitherSkull
AncientWarfareNpc.native.civilian.female
GrimoireOfGaia.Swamper
BiomesOPlenty.JungleSpider
chocolateQuest.Lich
lotr.DwarfAxeThrower
AncientWarfareNpc.pirate.civilian.male
lotr.UrukHaiTrader
fossil.Failuresaurus
lotr.BlueDwarfWarrior
npcelfmale
RopesPlus.RopeArrow
lotr.UtumnoObsidianWarg
lotr.DolGuldurOrc
lotr.Termite
lotr.Troll
AncientWarfareNpc.pirate.cavalry
witchery.lordoftorment
GrimoireOfGaia.Shaman
Villager
GrimoireOfGaia.Baphomet
fossil.Stegosaurus
AncientWarfareNpc.native.archer
npcnagafemale
primitivemobs.MotherSpider
GrimoireOfGaia.Minotaurus
MoCreatures.Ent
lotr.UrukHaiMercenaryCaptain
MoCreatures.Snail
ProjectZulu|Core.Yellow Finch
ProjectZulu|Core.Beaver
lotr.DwarfWarrior
MoCreatures.Raccoon
ProjectZulu|Core.Alligator
witchery.demon
ProjectZulu|Core.Goat
MoCreatures.Fishy
AncientWarfareNpc.desert.archer.elite
witchery.lilith
ProjectZulu|Core.Mimic
lotr.MirkwoodSpider
ProjectZulu|Core.Penguin
lotr.WoodElfWarrior
AncientWarfareNpc.aw_npc_trader
lotr.BlueDwarfCommander
GrimoireOfGaia.FutakuchiOnna
ProjectZulu|Core.BeetleAS
Spider
lotr.GaladhrimLord
lotr.AngmarOrc
BiomesOPlenty.Pixie
AncientWarfareNpc.native.soldier
lotr.BlueDwarfAxeThrower
MoCreatures.Turkey
AncientWarfareNpc.pirate.soldier
AncientWarfareNpc.custom_1.archer.elite
lotr.TormentedElf
fossil.Coelacanth
lotr.AngmarOrcBombardier
fossil.Sarcosuchus
ProjectZulu|Core.Mummy
GSSkullCrawler
ProjectZulu|Core.Black Bear
lotr.HobbitFarmer
lotr.BlueDwarfMerchant
chocolateQuest.spaceWarrior
GSWitherSkullCrawler
witchery.witchhunter
MoCreatures.DragonFly
AncientWarfareNpc.pirate.priest
ProjectZulu|Core.Follower
lotr.Huorn
lotr.UrukWargBombardier
GrimoireOfGaia.Cyan Flower
Savanna
lotr.UrukHaiBannerBearer
doggystyle.Dog
lotr.RangerIthilienCaptain
witchery.wingedmonkey
fossil.Mosasaurus
GrimoireOfGaia.Slime Girl
chocolateQuest.TurtleBossPart
AncientWarfareNpc.bandit.archer.elite
Creeper
GrimoireOfGaia.Mermaid
lotr.UtumnoTroll
fossil.Ankylosaurus
witchery.hornedHuntsman
smallboats.EntityBoatPart
fossil.Velociraptor
primitivemobs.LilyLurker
lotr.HalfTroll
EyeOfEnderSignal
GrimoireOfGaia.Dhampir
MoCreatures.Snake
lom_Mole
AncientWarfareNpc.pirate.bard
fossil.Triceratops
lotr.MarshWraith
lom_SaberTooth
MoCreatures.Scorpion
primitivemobs.SupportCreeper
AncientWarfareNpc.custom_3.archer.elite
chocolateQuest.necromancer
AncientWarfareNpc.custom_1.leader
AncientWarfareNpc.bandit.leader
chocolateQuest.bull
lotr.JungleScorpion
witchery.louse
chocolateQuest.armoredSkeleton
Cow
lotr.GondorTowerGuard
lotr.GundabadBannerBearer
lotr.Bandit
lotr.GondorBannerBearer
AncientWarfareNpc.native.cavalry
AncientWarfareNpc.native.soldier.elite
GrimoireOfGaia.Cockatrice
Giant
witchery.banshee
lotr.GondorArcher
fossil.Pterosaur
npcEnderman
lotr.DunlendingDrunkard
lotr.DarkHuorn
MoCreatures.ButterFly
ProjectZulu|Core.Monkey
Monster
ProjectZulu|Core.Rabbit
witchery.goblingulg
primitivemobs.BewitchedTome
fossil.Gallimimus
witchery.mandrake
NpcSlime
GrimoireOfGaia.Mandragora
MoCreatures.HellRat
MoCreatures.Turtle
lotr.MordorOrcMercenaryCaptain
lotr.NearHaradPlantsTrader
AncientWarfareNpc.pirate.trader
GrimoireOfGaia.Satyr
MoCreatures.Cricket
AncientWarfareNpc.viking.archer
GrimoireOfGaia.Naga
witchery.hellhound
lotr.OlogHai
weaponmod.spear
ProjectZulu|Core.Deer
lotr.Camel
lotr.NearHaradBannerBearer
lotr.HighElfWarrior
lotr.BlueDwarfBannerBearer
witchery.spirit
witchery.covenwitch
lotr.NearHaradrimWarrior
lotr.RohanBarrowWraith
lotr.UtumnoFireWarg
chocolateQuest.mecha
EnderDragon
lotr.MordorOrcSlaver
lotr.GundabadWarg
ProjectZulu|Core.Armadillo
npczombiefemale
ProjectZulu|Core.TreeEnt
primitivemobs.Chameleon
GSSkeletonCat
AncientWarfareNpc.custom_3.trader
chocolateQuest.slimeBoss
lotr.HobbitShirriff
lotr.GundabadOrc
chocolateQuest.giantZombie
AncientWarfareNpc.custom_2.civilian.male
MoCreatures.BigGolem
AncientWarfareNpc.pirate.leader
GrimoireOfGaia.Creeper Girl
AncientWarfareNpc.custom_2.trader
fossil.Plesiosaur
lom_Ichtyosaurus
lotr.DwarfMiner
ExtendedFamPack.Familiar Rana
npczombiemale
GrimoireOfGaia.Cobble Golem
lotr.HalfTrollScavenger
GrimoireOfGaia.Banshee
Ozelot
Griseus
GrimoireOfGaia.Mimic
lom_Brontosaurus
AncientWarfareNpc.custom_2.leader
ExtendedFamPack.Familiar Fairy
witchery.nightmare
ProjectZulu|Core.Horn Bill
lotr.UrukHai
lotr.HighElfLord
GrimoireOfGaia.Sludge Girl
GrimoireOfGaia.Hunter
MoCreatures.Roach
lotr.MordorOrcBombardier
ProjectZulu|Core.Boar
lotr.Saruman
GrimoireOfGaia.Yeti
fossil.Dilophosaurus
AncientWarfareNpc.pirate.mounted_archer
lotr.RohanMeadhost
lotr.Elk
AncientWarfareNpc.custom_3.soldier
lotr.BlueDwarf
fossil.Bones
lotr.RangerIthilienBannerBearer
lotr.NearHaradrim
ProjectZulu|Core.OstrichEgg
ProjectZulu|Core.CreeperBlossomPrimed
smallboats.EntityBoatChest
lotr.AngmarWargBombardier
lotr.GundabadOrcArcher
lotr.GondorSoldier
lotr.NearHaradFoodTrader
primitivemobs.LostMiner
EntityHorse
CrocMonitor
GrimoireOfGaia.Sharko
lotr.Butterfly
AncientWarfareNpc.bandit.bard
AncientWarfareNpc.desert.civilian.male
AncientWarfareNpc.native.leader
npchumanmale
chocolateQuest.Referee
AncientWarfareNpc.aw_npc_courier
lotr.DolGuldurOrcArcher
lotr.BlackUruk
ExtendedFamPack.Familiar 1UP
lotr.GundabadOrcMercenaryCaptain
GSZombieDog
lotr.Zebra
chocolateQuest.abyssWalkerBoss
AncientWarfareNpc.custom_2.mounted_archer
lotr.MountainTroll
fossil.Mammoth
AncientWarfareNpc.desert.trader
lotr.MordorOrc
lotr.NurnSlave
AncientWarfareNpc.custom_3.civilian.female
WitherBoss
ProjectZulu|Core.SandWorm
lotr.MordorOrcSpiderKeeper
fossil.PigBoss
witchery.toad
Witch
lotr.Midges
fossil.Smilodon
AncientWarfareNpc.pirate.civilian.female
lotr.DunlendingWarlord
npccrystal
lotr.StoneTroll
GrimoireOfGaia.NineTails
lotr.AngmarOrcArcher
GrimoireOfGaia.Creep
ProjectZulu|Core.Pelican
lotr.Ent
MoCreatures.Bird
GrimoireOfGaia.Butler
lom_MuskOx
MoCreatures.Crocodile
witchery.deathhorse
lotr.RangerIthilien
lom_Pterosaurus
ProjectZulu|Core.Skeletonn
npcelffemale
ProjectZulu|Core.Blue Finch
lotr.HobbitFarmhand
MoCreatures.Ostrich
MoCreatures.MiniGolem
fossil.FriendlyPigZombie
lotr.Crebain
witchery.wolfman
primitivemobs.SkeletonWarrior
fossil.TerrorBird
witchery.babayaga
AncientWarfareNpc.custom_3.archer
primitivemobs.TravelingMerchant
lotr.MordorOrcTrader
AncientWarfareNpc.custom_1.soldier.elite
SnowMan
GrimoireOfGaia.Ender Eye
chocolateQuest.specterBoss
Croc
witchery.attackbat
MoCreatures.SilverSkeleton
fossil.Brachiosaurus
Mob
AncientWarfareNpc.custom_2.priest
ExtendedFamPack.Familiar Navi
AncientWarfareNpc.custom_2.civilian.female
lotr.UtumnoIceSpider
ProjectZulu|Core.Frog
AncientWarfareNpc.custom_1.archer
AncientWarfareNpc.custom_1.civilian.male
AncientWarfareNpc.viking.civilian.male
ProjectZulu|Core.Mummy Pharaoh
chocolateQuest.pigzombie
ProjectZulu|Core.FishB
GrimoireOfGaia.Spriggan
ProjectZulu|Core.Rhino
ProjectZulu|Core.FishA
primitivemobs.Summoner
npcnagamale
MoCreatures.Bee
Ghast
AncientWarfareNpc.custom_1.priest
lotr.NearHaradMineralsTrader
MushroomCow
lotr.Gollum
AncientWarfareNpc.desert.archer
lotr.Horse
MoCreatures.Fox
GrimoireOfGaia.Minotaur
lotr.HalfTrollBannerBearer
lotr.HalfTrollWarlord
Lace
GSSkeletonDog
lotr.Rhino
MoCreatures.Bunny
chocolateQuest.gremlin
fossil.Compsognathus
AncientWarfareNpc.custom_3.priest
DesertTortoise
GrimoireOfGaia.Succubus
lotr.WoodElfScout
fossil.AncientJavelin
ProjectZulu|Core.Ostrich
lotr.UrukHaiCrossbower
lotr.BlackUrukArcher
BiomesOPlenty.Rosester
witchery.goblin
npcGolem
MoCreatures.WWolf
lotr.Hobbit
lotr.DwarfCommander
lotr.RohirrimArcher
AncientWarfareNpc.custom_1.leader.elite
lotr.RangerNorthBannerBearer
Alligator
witchery.cat
lotr.UtumnoWarg
Sheep
AncientWarfareNpc.custom_3.mounted_archer
lotr.AngmarBannerBearer
AncientWarfareNpc.aw_npc_bard
MoCreatures.Wraith
witchery.leonard
Perentie
lotr.HalfTrollWarrior
lotr.MountainTrollChieftain
lom_WinterDeer
Blaze
AncientWarfareNpc.desert.priest
witchery.ent
CaveSpider
AncientWarfareNpc.desert.mounted_archer
npcenderchibi
AncientWarfareNpc.pirate.leader.elite
AncientWarfareNpc.native.leader.elite
ProjectZulu|Core.Red Finch
primitivemobs.TreasureSlime
Iguana
AncientWarfareNpc.desert.civilian.female
AncientWarfareNpc.custom_1.bard
MoCreatures.WildHorse
lom_Crab
lotr.Balrog
lotr.GondorBlacksmith
MoCreatures.Dolphin
ProjectZulu|Core.Crow
lotr.Lioness
ProjectZulu|Core.Vulture
witchery.lostsoul
ExtrabiomesXL.scarecrow
AncientWarfareNpc.viking.soldier.elite
MoCreatures.Fly
AncientWarfareNpc.bandit.mounted_archer
witchery.vampire
MoCreatures.Shark
chocolateQuest.SummonedUndead
lotr.Rabbit
ProjectZulu|Core.Giraffe
AncientWarfareNpc.native.civilian.male
witchery.death
lotr.NearHaradrimArcher
BiomesOPlenty.Phantom
AncientWarfareNpc.pirate.soldier.elite
lotr.DikDik
fossil.Nautilus
MoCreatures.Goat
lotr.Flamingo
GrimoireOfGaia.Gryphon
chocolateQuest.turtleBoss
Chameleon
lotr.Giraffe
GrimoireOfGaia.Dryad
ProjectZulu|Core.Duck
lotr.HaradPyramidWraith
AncientWarfareNpc.viking.bard
AncientWarfareNpc.bandit.civilian.male
MoCreatures.HorseMob
AncientWarfareNpc.custom_2.cavalry
ProjectZulu|Core.Gorilla
AncientWarfareNpc.viking.priest
GrimoireOfGaia.Bone Knight
MoCreatures.BigCat
chocolateQuest.armoredZombie
lotr.GondorianCaptain
GrimoireOfGaia.Werecat
MoCreatures.Ant
lotr.WoodElfCaptain
AncientWarfareNpc.pirate.archer.elite
AncientWarfareNpc.desert.bard
lom_Whale
lotr.DolGuldurChieftain
lotr.DolGuldurOrcTrader
MoCreatures.FlameWraith
lotr.WoodElfBannerBearer
witchery.darkmark
LargeCroc
lotr.MarshWraithBall
BiomesOPlenty.Wasp
lotr.MysteryWeb
MoCreatures.Werewolf
GrimoireOfGaia.Witch
npcpony
lotr.Bird
witchery.poltergeist
witchery.treefyd
ProjectZulu|Core.Fox
ProjectZulu|Core.Camel

**Comment by GotoLink** [comment]
All suggested settings have been added.
Listing is a duplicate of #254
